enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,170 @@
+
+## Motivation
+
+An initial driver of this feature are the centralized host management use cases
+in edge topologies, which without this feature, is restricted to having the
+central OpenShift cluster deployed on baremetal.
+
+### Goals
+
+The specific goals of this proposal are to:
+
+Support the ZTP (Zero Touch Provisioning) use case by partially enabling Baremetal Host API
+on the following platforms:
+- None
+- OpenStack
+- vSphere
+
+We will be successful when:
+
+ZTP can deploy clusters when running on the above platforms.
+
+### Non-Goals
+
+Allow Baremetal Host API to be fully enabled on all platforms.
+
+Note that the scope of this epic is only enablement of the Metal3
+BaremetalHost API and associated controller/services - the edge ZTP flow
+currently interacts directly via this API without any Machine API integration.
+
+## Proposal
+
+BMO (baremetal-operator) provides the Baremetal Host API, it in turn is configured
+and managed by CBO (cluster-baremetal-operator).
+
+CBO reads the Provisioning CR that is created by the installer on baremetal platforms
+and uses that to configure and deploy BMO.
+
+Currently CBO checks the platform and if it is not baremetal it will be in a "disabled" state i.e. it will
+1. set status.conditions Disabled=true and
+2. not read or process the Provisioning CR and thus not deploy baremetal-operator.
+
+This proposal is to allow CBO to be enabled on the Platforms: None, OpenStack or vSphere.
+
+Further (to restrict the testing matrix) the allowed configuration options
+of the Provisioning CR will be restricted to exactly those required by ZTP.
+
+*Only spec.provisioningNetwork=Disabled mode will be accepted in the Provisioning CR.*
+
+If any other provisioningNetwork mode is set, the CBO webhook will refuse the change.
+
+### User Stories
+
+#### Story 1 - Current IPI baremetal platform use case
+
+No change.
+
+#### Story 2 - ZTP use case
+
+As a user of a hub cluster that performs central infrastructure management, and
+optionally zero-touch provisioning, I need to provision hosts using the k8s-native
+API (Baremetal Hosts CR) even when the hub cluster has a platform of None, OpenStack, or vSphere.
+
+
+### Risks and Mitigations
+
+There is concern that *random* customers will use this feature out of context
+and create support burden. This is why we have not suggested enabling CBO on
+all platforms and with full feature set. However it is still a potential issue.
+
+## Design Details
+
+### Open Questions [optional]
+
+Re ZTP requirements:
+1. Are these all the platforms that are required?
+2. Is just enabling spec.provisioningNetwork=Disabled mode sufficient?
+
+### Test Plan
+
+#### Unit Testing
+
+We will add unit tests to confirm that cluster-baremetal-operator:
+* is enabled on the required platforms.
+* will restrict functionality on these platforms to ProvisioningNetork=Disabled.
+
+#### e2e Testing
+
+It is expected that these use cases will be functionally tested by the Assisted Installer
```

Comment:
I do expect Metal to have a CI verifying that the basic functionality works - outside of ZTP scope. AI is agnostic to a platform it runs on, hence AI team can not own multiple platforms to test metal deployment.
Of course it can be done under ZTP ci umbrella, but the expectation on developing the CI part is on the owner of the enhancement.

Reply:
I understand, however this seems to me to be one small part of a much bigger solution that will get functionally tested - which is why I put that statement in.. 
So what system is creating the Provisioning CR? that might be a good starting point to figuring out where the e2e belongs.

Reply:
I don't see a need for e2e testing of metal on every platform, as long as the test uses a configuration that would be compatible with what's being made available on platforms other than baremetal. I think that means configuring the test environment to not use a provisioning network, right?

Reply:
Not supporting a provisioning network certainly simplifies things and reduces the complexity of testing on additional platforms, but we probably do still need test coverage *somewhere* of every platform we expect to support (even if it's just downstream QE validation of the ZTP flow).

Reply:
thanks all, I have updated this now.

Reply:
I think this thread can be resolved.